src/CoreShop/Payum/HeidelpayBundle/Extension/PopulateHeidelpayExtension.php

```diff
@@ -99,24 +101,32 @@ public function onPostExecute(Context $context)
 
         $result['language'] = $gatewayLanguage;
         $result['customer'] = $customerData;
-        $result['basket']['amount'] = round($result['basket']['amount'] / 100, 2);
-
-        $request->setResult((array)$result);
+        $result['basket']['amount'] = $this->calcAmount($result['basket']['amount']);
```

Comment:
i think $payment->getCurrencyCode() can be used here as second parameter, otherwise it's fixed to EUR

Reply:
Makes in this case not a really difference (we just format the amount value), but you are right. I've added this :-)